rtt_ros2_params/README.md

```diff
@@ -0,0 +1,49 @@
+
+# Description
+
+ROS2 provides a slightly different infrastructure to deal with parameters. An introduction guide can be found in [Understanding ROS2 prameters](https://index.ros.org/doc/ros2/Tutorials/Parameters/Understanding-ROS2-Parameters/).
+The main difference is that there is no longer a centralized parameter server, but instead, parameters are maintained per node.
+
+## `ros2-params` service
+
+This PR provides a new Orocos service called `ros2-params` that can be loaded globally and per Orocos component. The service provides 4 operations:
+* `getParameter()`: loads a parameter from the ROS2 node parameter facility and returns its value.
+* `loadProperty()`:  loads a parameter from the ROS2 node parameter facility into an Orocos property. If the property doesn't exist, in creates a new one owned by the `Params` service.
+* `setParameter()`: sets a parameter into the ROS2 node parameter facility. If the parameter doesn't exists, it declares a new one.
+* `storeProperty()`: sets a parameter into the ROS2 node parameter facility from the value of an Orocos property. If the parameter doesn't exists, it declares a new one.
+
+## Usage
+
+This package can be imported with:
+```
+import("rtt_ros2_params")
+```
+
+Then, the service can be loaded into a component named `<component>` with:
+```
+<component>.loadService('ros2-params')
+```
+
+`<component>` will have a new service named `Params` that provides the operations described. These operations can be called with:
+```
+<component>.Params.setParameter( <args> )
+<component>.Params.getParameter( <args> )
+<component>.Params.loadProperty( <args> )
+<component>.Params.storeProperty( <args> )
+```
+
+## Requriements
+
+In order to work, this service requires two other elements being loaded before calling the operations.
+* A ROS2 node service must exist for the component where this service is loaded or globally.
+* The typekits provided in `rtt_ros2_rclcpp_typekit` and its dependencies must be loaded
+
+These prerequirements can be achieved by:
+```
+import("rtt_ros2")
+import("rtt_ros2_node")
+ros.import("rtt_ros2_rclcpp_typekit")
+
+# Optionally
+<component>.loadService("ros2_node")
```

Comment:
But, actually, it was not needed to get and set the parameters. I never used this `ros.create_node()` when testing this PR.

Reply:
For a component loading the service like that is sufficient and the service constructor will also create the `rclcpp::Node` instance. Only the global node needs to be instantiated explicitly. What I meant is that you need either

```
# Option 1: Instantiate a ROS node for a single component (with name /component_name)
<component>.loadService("ros2_node")
```
or
```
# Option 2: Instantiate a global, process-wide ROS node (with name /rtt_some_random_number)
ros.create_node()
```

But you cannot omit it completely, so it is not optional as the comment suggests. Otherwise [get_ros2_node()](https://github.com/orocos/rtt_ros2_integration/pull/7/files/ab4c541aad7cd022446fc8df0d09e630bfc7b869#diff-bdc8e8b6c7c11afe11fa3688d4ccfa59R268) would fail (to be replaced by the helper methods provided by package `rtt_ros2_node` itself).

What we could do is to also load the `ros2_node` service implicitly into a component whenever the `ros2-params` service is, or to create a global node implicitly on first use of the `ros2-params` service.